Sources/StreamChat/APIClient/Endpoints/Payloads/ChannelListPayload.swift

```diff
@@ -58,10 +58,10 @@ extension ChannelPayload: Decodable {
             channel: try container.decode(ChannelDetailPayload.self, forKey: .channel),
             watcherCount: try container.decodeIfPresent(Int.self, forKey: .watcherCount),
             watchers: try container.decodeIfPresent([UserPayload].self, forKey: .watchers),
```

Comment:
We should introduce a `decodeArrayIfPresentIgnoringFailures`
`decodeIfPresent` is not enough because it guards us against it being nil, but whenever it is not nil, it fails the same way as a regular decode.

Reply:
Let's add tests to cover for this before introducing it. Make it fail, and then introduce `decodeArrayIfPresentIgnoringFailures`

Reply:
I'll open a new PR for this, while approved let's keep the scope small & get it merged